### PR TITLE
Update "NFT Added" events

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

We want to have a better understanding of how NFT tracking features are used on MetaMask. This branch updates the events we fire for tracking when an NFT is added by a user to align with the expected segment schema. Can be tested using the [mock segment server](https://github.com/MetaMask/metamask-extension/blob/develop/development/README.md#debugging-with-the-mock-segment-api) or inspecting extension networking debug tab in browser

EDIT: also updates TokenAdded events to align with the segment schema

See: [ticket](https://github.com/MetaMask/MetaMask-planning/issues/678) and [related segment-schema PR](https://github.com/Consensys/segment-schema/pull/72)

### After
Screenshot 2023-06-27 at 4 21 23 PM

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/add-nft/add-nft.js [hunk 1]
```diff
@@ -100,19 +101,19 @@ export default function AddNft() {
...
-      event: MetaMetricsEventName.TokenAdded,
-      category: 'Wallet',
+      event: MetaMetricsEventName.NftAdded,
+      category: MetaMetricsEventCategory.Wallet,
       sensitiveProperties: {
         token_contract_address: nftAddress,
         token_symbol: tokenDetails?.symbol,
-        tokenId: tokenId.toString(),
-        asset_type: AssetType.NFT,
+        token_id: tokenId.toString(), // This should already be a string?
```
Comment (jiexi) [inline ui/pages/add-nft/add-nft.js:113]: does this need to be `.toString()`ed?

Comment (adonesky1) [reply ui/pages/add-nft/add-nft.js:113]: Looks like you're right, but could you verify by testing it manually just to be absolutely sure before we remove those `toString()`s?

**Comment by adonesky1** [comment]
@jiexi would you mind changing `source_connection_method` to `source` for all of the `MetaMetricsEventName.TokenAdded` events while we're at it since it seems to not match the [schema](https://github.com/Consensys/segment-schema/blob/e754c9ca7fd4050a1ad96e63dabb783be486c510/libraries/events/metamask-account-mgmt/token-added.yaml#L13C3-L13C9)?

File: ui/pages/add-nft/add-nft.js (hunk 1 above)
Comment (jiexi) [reply ui/pages/add-nft/add-nft.js:113]: verified manually. not needed

File: ui/pages/confirm-import-token/confirm-import-token.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ const ConfirmImportToken = () => {
           token_contract_address: pendingToken.address,
           token_decimal_precision: pendingToken.decimals,
           unlisted: pendingToken.unlisted,
-          source_connection_method: pendingToken.isCustom
+          source: pendingToken.isCustom
```
Comment (adonesky1) [inline ui/pages/confirm-import-token/confirm-import-token.js:54]: I know this isn't really part of the changes you set out to make but it looks like `isCustom` is a boolean? I'm not quite sure what we want it to be if isCustom is false. But for now seems like we should do:

```
pendingToken.isCustom ? MetaMetricsTokenEventSource.Custom : ''
```

**Review by adonesky1 — DISMISSED** [review]
LGTM

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by shanejonas — APPROVED** [review]

**Pull request merged**